api/test/wfl/integration/sinks/workspace_sink_test.clj

```diff
@@ -79,44 +80,59 @@
-                              {:called-with args})))]
-      (with-redefs-fn
-        {#'rawls/batch-upsert        verify-upsert-request
-         #'sink/entity-exists?       (constantly false)
-         #'firecloud/delete-entities (partial throw-if-called "delete-entities")}
-        #(sink/update-sink! executor sink))
-      (is (zero? (stage/queue-length executor)) "The workflow was not consumed")
-      (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-        (let [[record & rest] (->> sink :details (postgres/get-table tx))]
-          (is record "The record was not written to the database")
-          (is (empty? rest) "More than one record was written")
-          (is (= (:uuid workflow) (:workflow record))
-              "The workflow UUID was not written")
-          (is (= testing-entity-name (:entity record))
-              "The entity was not correct"))))))
+  (letfn [(sink [identifier]
+            (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+              (->> {:name           "Terra Workspace"
+                    :workspace      "workspace-ns/workspace-name"
+                    :entityType     testing-entity-type
+                    :fromOutputs    (resources/read-resource
+                                     "sarscov2_illumina_full/entity-from-outputs.edn")
+                    :identifier     identifier
+                    :skipValidation true}
+                   (sink/create-sink! tx (rand-int 1000000))
+                   (zipmap [:sink_type :sink_items])
+                   (sink/load-sink! tx))))
+          (verify-upsert-request [workspace [[type name _]]]
+            (is (= "workspace-ns/workspace-name" workspace))
+            (is (= type testing-entity-type))
+            (is (= name testing-entity-name)))
+          (throw-if-called [fname & args]
+            (throw (ex-info (str fname " should not have been called")
+                            {:called-with args})))]
+    (let [workflow     {:uuid    "2768b29e-c808-4bd6-a46b-6c94fd2a67aa"
+                        :status  "Succeeded"
+                        :outputs (-> "sarscov2_illumina_full/outputs.edn"
+                                     resources/read-resource
+                                     (assoc :flowcell_id testing-entity-name))}
+          executor     (make-queue-from-list [[nil workflow]])
+          sink-throws  (sink "not-a-workflow-input-or-output")
+          sink-updates (sink "flowcell_id")]
+      (testing "Sink identifier matches no workflow output or input"
+        (with-redefs-fn
+          {#'rawls/batch-upsert        (partial throw-if-called "rawls/batch-upsert")
+           #'sink/entity-exists?       (partial throw-if-called "sink/entity-exists?")
+           #'firecloud/delete-entities (partial throw-if-called "firecloud/delete-entities")}
+          #(is (thrown-with-msg?
+                ExceptionInfo (re-pattern sink/entity-name-not-found-error-message)
```

Comment:
If you took my prior suggestion, then this should be:
```clojure
ExceptionInfo (-> sink/entity-name-not-found-error-message
                  var var-get re-pattern)
```
or something.

Reply:
(Doesn't really matter but) wouldn't this work?

`ExceptionInfo (re-pattern (sink/entity-name-not-found-error-message))`

Reply:
Whoops, bad reading comprehension.  Disregard.